tests/test_factory.py

```diff
@@ -869,6 +869,19 @@ def test_create_poetry_with_invalid_dev_dependencies(caplog: LogCaptureFixture)
     assert any("dev" in r.groups for r in poetry.package.all_requires)
 
 
+def test_create_poetry_with_invalid_dependency_groups() -> None:
+    with pytest.raises(RuntimeError) as e:
+        _ = Factory().create_poetry(
+            fixtures_dir / "project_with_invalid_dependency_groups", with_groups=False
```

Comment:
Since you explicitly passed `with_groups=False`, I am wondering if this should raise an error if `with_groups` is `False` or only if `with_groups` is `True`?

`with_groups=False` is used for building, i.e. when dependency groups are irrelevant. On the other side, violating the scheme is so essential it may be fine to prevent building.

Anyway, I would make the test parametrized with `with_groups` `False` and `True`.

Reply:
> On the other side, violating the scheme is so essential it may be fine to prevent building.

I agree. I parametrized the test.